packages/components/addon/components/hds/pagination-bar/index.hbs

```diff
@@ -1,70 +1,31 @@
-<div class={{this.classNames}} ...attributes>
-  {{#if this.showTotalCount}}
-    <Hds::PaginationBar::PaginationTotalCount @hasTotalItems={{this.hasTotalItems}} />
-  {{/if}}
-
-  <Hds::PaginationBar::Pagination @type={{this.type}} />
-
-  {{#if this.showPageSize}}
-    <Hds::PaginationBar::PaginationPageSize />
-  {{/if}}
-</div>
-
-{{!
-  TEMP ROUGH NOTES
-  ------------------
-  Figuring out class naming and DOM structure...
-
-  WIP Structure (some names represent sub-components, others just represent HTML/CSS structure):
-    * PaginationBar
-      * PaginationTotalCount (optional)
-
-      * Pagination
-        * page-prev
-        * pages
-          * page
-            * page-link (some content aren't links though...)
-        * page-next
-
-      * PaginationPageSize (optional)
-
-APIs
-------
-PaginationBar:
-  X type=
-    "numbered" (default)
-    "compact"
-  X showTotalCount ?) =
-    true (default)
-    false
-  X showPageSize =
-    true (default)
-    false
-  isTruncated =
-    true (default ?)
-    false
-  X hasTotalItems=
-    true (default)
-    false
-
-TotalCount:
-  X hasTotalItems= (take values passed down from PaginationBar unless used as stand alone component)
-    true (default)
-    false
-
-Pagination:
-  X type= (take values passed down from PaginationBar unless used as stand alone component)
-    "numbered" (default)
-    "compact"
-  isTruncated= (take values passed down from PaginationBar unless used as stand alone component)
-    true (default ?)
-    false
-  * PageNumber:
-      isSelected (NOT isCurrent) (for page number, not exposed)
-      Truncated/ellipsis ? (for page number, not exposed)
-  * Prev/Next buttons:
-    * isDisabled (not exposed)
-
-PageSize:
-  ?
-}}
\ No newline at end of file
+<div class="hds-pagination-bar" ...attributes>
+  {{yield
+    (hash
+      TotalCount=(component
+        "hds/pagination-bar/total-count"
+        hasTotalItems=@hasTotalItems
+        totalItems=@totalItems
+        currentPage=this.currentPage
+        itemsPerPage=this.itemsPerPage
+      )
+    )
+  }}
+
+  {{yield
+    (hash
+      Pagination=(component
+        "hds/pagination-bar/pagination"
+        type=this.type
+        totalPages=this.totalPages
+        currentPage=this.currentPage
+        onPageChange=this.onPageChange
+      )
+    )
+  }}
+
+  {{yield
+    (hash
+      PageSize=(component "hds/pagination-bar/page-size" onChange=this.onPageSizeChange itemsPerPage=this.itemsPerPage)
```

Comment:
super nitpick: in general we tend to have arguments/props first, and then event handlers
```suggestion
      PageSize=(component "hds/pagination-bar/page-size" itemsPerPage=this.itemsPerPage onChange=this.onPageSizeChange)
```

for example: https://github.com/hashicorp/design-system/blob/main/packages/components/addon/components/hds/tabs/index.hbs#L9-L14

Reply:
Thanks, I still need to clean up and reorganize some of the code like this but this is helpful.